src/DesignQuery/DesignQuery.cpp

```diff
@@ -57,7 +57,113 @@ extern FOEDAG::Session* GlobalSession;
+  std::filesystem::path dir = GetProjDir();
+  std::filesystem::path port_info = "port_info.json";
+  return dir / port_info;
+}
 
 bool DesignQuery::RegisterCommands(TclInterpreter* interp, bool batchMode) {
+  auto parse_design_data = [](void* clientData, Tcl_Interp* interp, int argc,
+                              const char* argv[]) -> int {
+    DesignQuery* design_query = (DesignQuery*)clientData;
+    Compiler* compiler = design_query->GetCompiler();
+    bool status = true;
+
+    json& hier_json = design_query->getHierJson();
+    json& port_json = design_query->getPortJson();
+
+    std::filesystem::path port_info_path = design_query->GetPortInfoPath();
+    std::filesystem::path hier_info_path = design_query->GetHierInfoPath();
+    if (!FileUtils::FileExists(port_info_path)) {
+      status = false;
+      compiler->Message(
+          "Unable to locate port_info.json in design directory: " +
+          design_query->GetProjDir().string());
+    } else {
+      std::ifstream port_info_f(port_info_path);
+      port_json = json::parse(port_info_f);
+    }
+    if (!FileUtils::FileExists(hier_info_path)) {
+      status = false;
+      compiler->Message(
+          "Unable to locate port_info.json in design directory: " +
+          design_query->GetProjDir().string());
+    } else {
+      std::ifstream hier_info_f(hier_info_path);
+      hier_json = json::parse(hier_info_f);
+    }
+
+    return (status) ? TCL_OK : TCL_ERROR;
+  };
+  interp->registerCmd("parse_design_data", parse_design_data, this, 0);
+
+  auto get_file_ids = [](void* clientData, Tcl_Interp* interp, int argc,
+                         const char* argv[]) -> int {
+    // DesignQuery* design_query = (DesignQuery*)clientData;
+    // Compiler* compiler = design_query->GetCompiler();
+    bool status = true;
+
+    return (status) ? TCL_OK : TCL_ERROR;
+  };
+  interp->registerCmd("get_file_ids", get_file_ids, this, 0);
+
+  auto get_modules = [](void* clientData, Tcl_Interp* interp, int argc,
+                        const char* argv[]) -> int {
+    // DesignQuery* design_query = (DesignQuery*)clientData;
+    // Compiler* compiler = design_query->GetCompiler();
+    bool status = true;
+
+    return (status) ? TCL_OK : TCL_ERROR;
+  };
+  interp->registerCmd("get_modules", get_modules, this, 0);
+
+  auto get_file_name = [](void* clientData, Tcl_Interp* interp, int argc,
+                          const char* argv[]) -> int {
+    // DesignQuery* design_query = (DesignQuery*)clientData;
+    // Compiler* compiler = design_query->GetCompiler();
+    bool status = true;
+
+    // Parse Argument file_id
+
+    return (status) ? TCL_OK : TCL_ERROR;
+  };
+  interp->registerCmd("get_file_name", get_file_name, this, 0);
+
+  auto get_top_module = [](void* clientData, Tcl_Interp* interp, int argc,
+                           const char* argv[]) -> int {
+    // DesignQuery* design_query = (DesignQuery*)clientData;
```

Comment:
Commented out for TODO later ?

e.g. https://github.com/os-fpga/FOEDAG/blob/main/tests/Testcases/IPGenerate/test_ipgenerate_modules.tcl#L22